File: VICSApp/CoreSystem/DHCD/FormUI/Meeting/AuthorizationListForSelect.cs
using System;
using System.Data;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class AuthorizationListForSelect
    {
        public AuthorizationListForSelect()
        {
            InitializeComponent();
            _DataGridView1.Name = "DataGridView1";
        }

        private string strHolderIdentifyCard;
        private string strDelegateIdentifyCard;

        public object HolderIdentifyCard
        {
            get
            {
                return strHolderIdentifyCard;
            }

            set
            {
                strHolderIdentifyCard = Conversions.ToString(value);
            }
        }

        public object DelegateIdentifyCard
        {
            get
            {
                return strDelegateIdentifyCard;
            }

            set
            {
                strDelegateIdentifyCard = Conversions.ToString(value);
            }
        }

        private void ToolStripButton4_Click(object sender, EventArgs e)
        {
            filldgv();
        }

        private void filldgv()
        {
            var dt = new DataTable();
            try
            {
                dt = My.MyProject.Forms.Mainform.BenlyDal.Authorizations_getlist(My.MyProject.Forms.Mainform.workingmeeting, 0m, "", strDelegateIdentifyCard, strHolderIdentifyCard);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
            }

            DataGridView1.DataSource = dt;
            ToolStripStatusLabel2.Text = DataGridView1.RowCount.ToString();
            int sumshares = 0;
            int sumvoterights = 0;
            foreach (DataRow dr in dt.Rows)
                // sumshares = sumshares + dr.Item("Shares")
                sumvoterights = Conversions.ToInteger(Operators.AddObject(sumvoterights, dr["Delegateright"]));
            // ToolStripStatusLabel4.Text = Mainform.addthousandseperator(sumshares.ToString)
            // ToolStripStatusLabel6.Text = Mainform.addthousandseperator(sumvoterights.ToString)
        }

        private void HolderList_Load(object sender, EventArgs e)
        {
            filldgv();
            // Me.MdiParent = Mainform
        }

        private void HolderListForSelect_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
            {
                Close();
            }
        }

        private void DataGridView1_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                e.SuppressKeyPress = true;
                Close();
            }
        }
    }
}

File: VICSApp/CoreSystem/DHCD/FormUI/Meeting/AuthorizationListForSelect.vb
Public Class AuthorizationListForSelect
    Dim strHolderIdentifyCard As String
    Dim strDelegateIdentifyCard As String

    Public Property HolderIdentifyCard()
        Get
            Return strHolderIdentifyCard
        End Get
        Set(ByVal value)
            strHolderIdentifyCard = value
        End Set
    End Property

    Public Property DelegateIdentifyCard()
        Get
            Return strDelegateIdentifyCard
        End Get
        Set(ByVal value)
            strDelegateIdentifyCard = value
        End Set
    End Property

    Private Sub ToolStripButton4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        filldgv()
    End Sub
    Private Sub filldgv()
        Dim dt As New DataTable
        Try
            dt = Mainform.BenlyDal.Authorizations_getlist(Mainform.workingmeeting, 0, "", strDelegateIdentifyCard, strHolderIdentifyCard)

        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
        End Try
        DataGridView1.DataSource = dt
        ToolStripStatusLabel2.Text = DataGridView1.RowCount
        Dim sumshares As Integer = 0
        Dim sumvoterights As Integer = 0

        For Each dr As DataRow In dt.Rows
            'sumshares = sumshares + dr.Item("Shares")
            sumvoterights = sumvoterights + dr.Item("Delegateright")
        Next
        'ToolStripStatusLabel4.Text = Mainform.addthousandseperator(sumshares.ToString)
        'ToolStripStatusLabel6.Text = Mainform.addthousandseperator(sumvoterights.ToString)
    End Sub


    Private Sub HolderList_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        filldgv()
        'Me.MdiParent = Mainform
    End Sub


    Private Sub HolderListForSelect_KeyUp(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyUp
        If e.KeyCode = Keys.Escape Then
            Me.Close()
        End If

    End Sub

    Private Sub DataGridView1_KeyDown(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles DataGridView1.KeyDown
        If e.KeyCode = Keys.Enter Then
            e.SuppressKeyPress = True
            Me.Close()
        End If
    End Sub
End Class